pkg/user/service.go

```diff
@@ -129,3 +130,24 @@ func (s service) FindOrCreate(email string, password string) (*model.User, error
 
 	return s.repository.findOrCreate(user)
 }
+
+func (s service) Update(id uint, email, password string) error {
+	user, err := s.repository.findById(id)
+	if err != nil {
+		return err
+	}
+
+	if email != "" {
+		user.Email = email
+	}
+
+	if password != "" {
+		var err error
+		user.Password, err = hashPassword(password)
+		if err != nil {
+			return fmt.Errorf("password hashing failed: %s", err)
+		}
+	}
+
+	return s.repository.update(user)
```

Comment:
Is the user object updated when calling the update method? I'm wondering if we might as well return it? You're reloading it in the handler

Reply:
Yep, seems like this is the better way to go about it. 👍